dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ManagedExecutorProducer.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.mp.cdi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorBuilder;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+public class ManagedExecutorProducer {
+
+    private static final TraceComponent tc = Tr.register(ManagedExecutorProducer.class);
+
+    // A new instance of this class will be constructed each time the producer is
+    // called so we need to keep the config->executor map in a static variable
+    private static final Map<ManagedExecutorConfigContainer, ManagedExecutor> configMap = new ConcurrentHashMap<>();
+
+    @Produces
+    public ManagedExecutor produceManagedExecutor(InjectionPoint point) {
+        ManagedExecutorConfig configAnno = point.getAnnotated().getAnnotation(ManagedExecutorConfig.class);
+        ManagedExecutorConfigContainer config = ManagedExecutorConfigContainer.get(configAnno);
+        ManagedExecutor exec = configMap.computeIfAbsent(config, this::createManagedExecutor);
+        return exec;
+    }
+
+    private ManagedExecutor createManagedExecutor(ManagedExecutorConfigContainer config) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+            Tr.debug(tc, "Building a new ManagedExecutor for the requested configuration: " + config);
+        return ManagedExecutorBuilder.instance()
+                        .maxAsync(config.maxAsync)
+                        .maxQueued(config.maxQueued)
+                        .propagated(config.propogated.toArray(new String[config.propogated.size()]))
```

Comment:
Doing this would require the main MP Concurrent bundle to export its impl package so the CDI producer could call the impl-specific builder method. Since these arrays will typically be small (1-4 elements) I don't think this is a big enough optimization to warrant exposing packages from the core MP Concurrent bundle

Reply:
I don't think we would do this by exporting an impl package. Instead, it would export an interface to our builder impl for this as well as other potential optimizations.  That said, I wouldn't bother making any of these changes until more discussion has happened in the spec to determine if this path is needed.